src/VisualStudio/Core/Def/Implementation/Diagnostics/DiagnosticProgressReporter.cs

```diff
@@ -21,180 +21,180 @@ internal sealed class TaskCenterSolutionAnalysisProgressReporter
-                // at the end, this notification will go away automatically
-                // but to make UI to be updated to right status eventually if task takes long time to finish
-                // we enqueue refresh task.
-                EnqueueRefresh();
-                return;
-            }
-
-            _lastShownProgressStatus = _lastProgressStatus;
-            _lastTimeReported = current;
-
-            ChangeProgress(_taskHandler, GetMessage());
-
-            string GetMessage()
-            {
-                var statusMessage = (_lastProgressStatus == ProgressStatus.Paused) ? ServicesVSResources.Paused_0_tasks_in_queue : ServicesVSResources.Evaluating_0_tasks_in_queue;
-                return string.Format(statusMessage, _lastPendingItemCount);
-            }
-
-            void EnqueueRefresh()
-            {
-                if (_resettableDelay != null)
-                {
-                    _resettableDelay.Reset();
-                    return;
-                }
-
-                _resettableDelay = new ResettableDelay((int)s_minimumInterval.TotalMilliseconds, AsynchronousOperationListenerProvider.NullListener);
-                _resettableDelay.Task.SafeContinueWith(_ =>
-                {
-                    _resettableDelay = null;
-                    ProgressUpdated();
-                }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
-            }
+            var statusMessage = (progressData.Status == ProgressStatus.Paused) ? ServicesVSResources.Paused_0_tasks_in_queue : ServicesVSResources.Evaluating_0_tasks_in_queue;
+            return string.Format(statusMessage, pendingItemCount);
         }
 
-        private void StartedOrStopped(bool started)
+        private void Started()
         {
-            if (started)
-            {
-                ResetProgressStatus();
+            _lastProgressData = default;
+            _lastProgressCount = 0;
 
-                // if there is any pending one. make sure it is finished.
-                _currentTask?.TrySetResult(default);
+            // Make sure when we start a new task, the previous one is removed from the task center.
+            _taskCenterTask?.TrySetResult(default);
 
-                var taskHandler = _taskCenterService.PreRegister(_options, data: default);
+            // Register a new task handler to handle a new task center task.
+            // Each task handler can only register one task, so we must create a new one each time we start.
+            var taskHandler = _taskCenterService.PreRegister(_options, data: default);
 
-                _currentTask = new TaskCompletionSource<VoidResult>();
-                taskHandler.RegisterTask(_currentTask.Task);
+            // Create a new non-completed task to be tracked by the task handler.
+            _taskCenterTask = new TaskCompletionSource<VoidResult>();
```

Comment:
this is all unsafe as nothing guarantees taht two StartCalls won't happen at the same time (in your PR currently).  can discuss ways to address this on gitter.